src/lib/logstore/log_dev.cpp

```diff
@@ -173,8 +173,8 @@ void LogDev::do_load(const off_t device_cursor) {
         const auto flush_ld_key =
             logdev_key{header->start_idx() + header->nrecords() - 1, group_dev_offset + header->total_size()};
         while (i < header->nrecords()) {
-            const auto* const rec{header->nth_record(i)};
-            const uint32_t data_offset{(rec->offset + (rec->get_inlined() ? 0 : header->oob_data_offset))};
+            const auto* rec = header->nth_record(i);
```

Comment:
Why are we converting all these initializer list based constructors to assignment style expressions? The former is the preferred style I believe.

Reply:
I believe it is for readability purpose (there is one discussion about this we had in one of the homestore meeting). `const auto` should be as good to cover the benefits we get from initializer list.

Reply:
We are not replacing initialized based constructors from {} to =, but we are replacing initialization of POD types to assignment as it doesn't provide any meaningful value and take a huge hit in readability in my opinion. It also doesn't clearly differentiate with the cases where `{}` is meaningful, for example in `std::unique_lock l{mtx}` which is a classic case where it is helpful.